wgpu-native/src/track.rs

```diff
@@ -112,6 +112,45 @@ pub enum Stitch {
     Last,
 }
 
+pub struct ConsumeIterator<'a, I: TypedId, U: Copy + PartialEq> {
```

Comment:
This looks nice :+1: I could see us using this in a few places in the future if we make it slightly more general (generic iterator and move the other fields into a generic context field)